library/pkparse.c

```diff
@@ -394,6 +396,10 @@ static int pk_group_id_from_group(const mbedtls_ecp_group *grp, mbedtls_ecp_grou
             mbedtls_mpi_cmp_mpi(&grp->G.Z, &ref.G.Z) == 0 &&
             /* For Y we may only know the parity bit, so compare only that */
             mbedtls_mpi_get_bit(&grp->G.Y, 0) == mbedtls_mpi_get_bit(&ref.G.Y, 0)) {
+
+            /* Free and load group by id */
+            mbedtls_ecp_group_free(grp);
```

Comment:
We need to free the members manually here, because `mbedtls_ecp_group_free()` won't free everything we need and there will be a leak. Not calling `mbedtls_ecp_group_free()` would be the best as eventually it will free even less.

Reply:
The freeing the individual memeber's logic has been moved to the `pk_group_id_from_specified()` clean-up.